src/parse.c

```diff
@@ -2677,19 +2675,24 @@ netplan_finish_parse(GError** error)
 {
     if (netdefs) {
         GError *recoverable = NULL;
+        GHashTableIter iter;
+        gpointer key, value;
         g_debug("We have some netdefs, pass them through a final round of validation");
         if (!validate_default_route_consistency(netdefs, &recoverable)) {
             g_warning("Problem encountered while validating default route consistency."
                       "Please set up multiple routing tables and use `routing-policy` instead.\n"
                       "Error: %s", (recoverable) ? recoverable->message : "");
             g_clear_error(&recoverable);
         }
-        g_hash_table_foreach(netdefs, finish_iterator, error);
-    }
-
-    if (error && *error)
-        return NULL;
+        g_hash_table_iter_init (&iter, netdefs);
 
+        while (g_hash_table_iter_next (&iter, &key, &value))
+        {
```

Comment:
```suggestion
        while (g_hash_table_iter_next (&iter, &key, &value)) {
```
Tiny code-style nitpick...